android/app/src/main/java/org/fossasia/openevent/adapters/GlobalSearchAdapter.java

```diff
@@ -0,0 +1,172 @@
+package org.fossasia.openevent.adapters;
+
+import android.content.Context;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import org.fossasia.openevent.R;
+import org.fossasia.openevent.data.Microlocation;
+import org.fossasia.openevent.data.Speaker;
+import org.fossasia.openevent.data.Track;
+import org.fossasia.openevent.viewholders.TrackViewHolder;
+import org.fossasia.openevent.viewholders.LocationViewHolder;
+import org.fossasia.openevent.viewholders.SpeakerViewHolder;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import io.reactivex.disposables.CompositeDisposable;
+
+/**
+ * Created by Shreyas on 5/15/2017.
+ */
+public class GlobalSearchAdapter extends BaseRVAdapter<Object,RecyclerView.ViewHolder>{
+
+    private Context context;
+    private CompositeDisposable disposable;
+
+    private List<Object> filteredResultList = new ArrayList<>();
+
+    //ViewType Constants
+    private final int TRACK = 0;
+    private final int SPEAKER = 2;
+    private final int LOCATION = 3;
+    private final int DIVIDER = 4;
+
+    public GlobalSearchAdapter(List<Object> dataList, Context context) {
+        super(dataList);
+        this.context = context;
```

Comment:
Use `context.getApplicationContext()` and store that here instead to avoid memory leaks